src/lib/array.jl

```diff
@@ -165,10 +167,21 @@ end
 # This is also used by comprehensions, which do guarantee iteration order.
 # Not done for pmap, presumably because all is lost if you are relying on its order.
 _tryreverse(m, backs, Δ) = backs, Δ
-_tryreverse(m::typeof(map), backs, Δ) = reverse(backs), reverse(Δ)
+_tryreverse(m::typeof(map), backs, Δ) = _reverse(backs), _reverse(Δ)
 
 _tryreverse(m, x) = x
-_tryreverse(m::typeof(map), x) = reverse(x)
+_tryreverse(m::typeof(map), x) = _reverse(x)
+
+# Fallback
+_reverse(x) = reverse(x)
+
+# Known cases in the standard library on which `reverse` errors (issue #1393)
+_reverse(x::LowerTriangular) = UpperTriangular(reverse(parent(x)))
+_reverse(x::UpperTriangular) = LowerTriangular(reverse(parent(x)))
+_reverse(x::UnitLowerTriangular) = UnitUpperTriangular(reverse(parent(x)))
+_reverse(x::UnitUpperTriangular) = UnitLowerTriangular(reverse(parent(x)))
+_reverse(x::Hermitian) = Hermitian(reverse(collect(x)))
+_reverse(x::Symmetric) = Symmetric(reverse(collect(x)))
```

Comment:
I guess we could avoid `collect` here, I think the following could work:
```suggestion
_reverse(x::Hermitian) = Hermitian(_reverse(x.data), x.uplo == 'U' ? :L : :U)
_reverse(x::Symmetric) = Symmetric(_reverse(x.data), x.uplo == 'U' ? :L : :U)
```

Reply:
Good point, thanks!

Reply:
BTW. this deviates from `reverse`, which makes a copy, but I think it makes sense here.

Reply:
This will make a copy as well, won't it? If `x.data isa Array`, then `_reverse(x.data) = reverse(x.data)` which will be a copy of `x.data`.

Reply:
Yeah that's right. I confused myself there. Never mind.